pkg/redact/literal.go

```diff
@@ -28,32 +31,36 @@ func (r literalRedactor) Redact(input io.Reader, path string) io.Reader {
 	go func() {
 		var err error
 		defer func() {
-			if err == io.EOF {
+			if err == nil || err == io.EOF {
 				writer.Close()
 			} else {
+				if err == bufio.ErrTooLong {
+					s := fmt.Sprintf("Error redacting %q. A line in the file exceeded %d MB max length", path, constants.SCANNER_MAX_SIZE/1024/1024)
+					klog.V(2).Info(s)
+				} else {
+					klog.V(2).Info(fmt.Sprintf("Error redacting %q: %v", path, err))
+				}
 				writer.CloseWithError(err)
 			}
 		}()
 
-		reader := bufio.NewReader(input)
+		buf := make([]byte, constants.BUF_INIT_SIZE)
+		scanner := bufio.NewScanner(input)
+		scanner.Buffer(buf, constants.SCANNER_MAX_SIZE)
+
 		lineNum := 0
-		for {
+		for scanner.Scan() {
 			lineNum++
-			var line string
-			line, err = readLine(reader)
-			if err != nil {
-				return
-			}
+			line := scanner.Bytes()
 
-			clean := strings.ReplaceAll(line, r.matchString, MASK_TEXT)
+			clean := bytes.ReplaceAll(line, r.match, maskTextBytes)
 
-			// io.WriteString would be nicer, but scanner strips new lines
-			fmt.Fprintf(writer, "%s\n", clean)
+			_, err = writer.Write(append(clean, '\n')) // Append newline since scanner strips it
```

Comment:
i think if the line is really long and you use `append(line, '\n')` you will allocate another up to 10M. wouldnt it just be better to write twice?

```suggestion
			_, err = writer.Write(clean)
			_, err = writer.Write('\n') // Append newline since scanner strips it
```

this goes for all occurrences of append below